File: C++/169. Majority Element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int len = nums.size();
        int cnt = 0,ans = 0;
        for(int i=0;i<len;i++)
        {
            if(ans==nums[i])
                cnt++;
            else
                cnt--;
            if(cnt<0)
            {
                cnt = 0;
                ans = nums[i];
            }
        }
        return ans;
    }
};

File: Python/169. Majority Element.py
class Solution:
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        ans, cnt = 0, 0
        for num in nums:
            if num == ans:
                cnt += 1
            else:
                cnt -= 1
            if cnt < 0:
                cnt = 0
                ans = num
        return ans